src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1004,6 +1004,13 @@ void Terminal::_AdjustCursorPosition(const COORD proposedPosition)
             _buffer->IncrementCircularBuffer();
             proposedCursorPosition.Y--;
             rowsPushedOffTopOfBuffer++;
+
+            // Update our selection too, so it doesn't move as the buffer is cycled
+            if (_selection)
+            {
+                _selection->start.Y -= 1;

```

Comment:
What if the selection starts to go off the top of the screen (Y=0)?

Reply:
Updated, if start.Y is at 0, we only reduce end.Y (if end.Y is greater than 0). If both are at 0, we clear the selection

Reply:
Update: if we're only reducing end.Y, then we also set start.X = 0 to keep the block selection